app/maps/mapVoteHandlers.js

```diff
@@ -0,0 +1,61 @@
+'use strict';
+
+import slackTextSnippets from '../slackTextSnippets';
+
+function _getMapPredicate(mapName) {
+    return (map) => {
+        return map.name === mapName;
+    };
+}
+
+function _getMapUpdateAction(keyword, voteStrategy, callback) {
+    return (map) => {
+        if (!map.votes) {
+            map.votes = {};
+        }
+
+        if (!map.votes[keyword]) {
+            map.votes[keyword] = 0;
+        }
+
+        map.votes[keyword] = voteStrategy(map.votes[keyword]);
+
+        callback(null, slackTextSnippets.decorate(map.name) + ' voted for ' + slackTextSnippets.decorate(keyword) + '; currently at ' + slackTextSnippets.decorate(map.votes[keyword]));
+    };
+}
+
+function _getMapVoteHandler(mapPersistence, voteStrategy) {
+    return {
+        makeItSo(parsedCommand, callback) {
+            if (parsedCommand.arguments.length < 2) {
+                callback(null, 'Map name and keyword is required for voting.');
+                return;
+            }
+
+            if (parsedCommand.arguments.length < 3) {
+                callback(null, 'Keyword is required for voting.');
```

Comment:
you're right, we could add suggestion with the full syntax in the message, like `upvotemap <mapname> <keyword>`